server/leafnode.go

```diff
@@ -51,6 +50,10 @@ const leafNodeReconnectAfterPermViolation = 30 * time.Second
 // Prefix for loop detection subject
 const leafNodeLoopDetectionSubjectPrefix = "$LDS."
 
+// Path added to URL to indicate to WS server that the connection is a
+// LEAF connection as opposed to a CLIENT.
+const leafNodeWSPath = "/leafnode"
```

Comment:
this means we can run clients and leafnodes on the same port like we discussed?

Reply:
Yep. The user will not need to put that in the url, only needs to have `ws://` to instruct the server to attempt a WS connection. The server will add the path to the url passed to the http.Request. The NATS Server (the internal web server that is running) will then look for a path in the incoming request's url and if it finds "/leafnode" will create the connection as a leafnode connection.
You gave me that idea when you said that a proxy could redirect to leafnode port if it found /leafnode in the url, and I took that a step further!